### ISSUE TITLE
GetService() error: service does not exist

### ISSUE THREAD

**Issue by Stono** [istio/istio#5997]
Hey, 
Istio: 0.8.0

I'm seeing these every minute in our pilot logs:

```
2018-06-03T15:52:36.612513Z     info    GetService(*.dev.k8.atcloud.io) error: service does not exist
2018-06-03T15:53:36.612222Z     info    GetService(*.dev.k8.atcloud.io) error: service does not exist
2018-06-03T15:53:36.612521Z     info    GetService(*.dev.k8.atcloud.io) error: service does not exist
2018-06-03T15:54:36.609401Z     info    GetService(*.dev.k8.atcloud.io) error: service does not exist
2018-06-03T15:54:36.612074Z     info    GetService(*.dev.k8.atcloud.io) error: service does not exist
```

I was really baffled where these are coming from, the only place where we specify that value is in this secret resource:

```
❯ kn get secret -o yaml atcloud-wildcard-ingress-cert
apiVersion: v1
data:
  tls.crt: omitted
  tls.key: omitted 
kind: Secret
metadata:
  annotations:
    certmanager.k8s.io/alt-names: '*.dev.k8.atcloud.io'
    certmanager.k8s.io/common-name: '*.dev.k8.atcloud.io'
    certmanager.k8s.io/issuer-kind: ClusterIssuer
    certmanager.k8s.io/issuer-name: letsencrypt-prod
  name: atcloud-wildcard-ingress-cert
  namespace: ingress-nginx
type: kubernetes.io/tls
```

Which is generated as part of this cert-manager certificate request:

```
❯ kn get certificate atcloud-wildcard -o yaml
apiVersion: certmanager.k8s.io/v1alpha1
kind: Certificate
metadata:
  clusterName: ""
  creationTimestamp: 2018-05-30T12:11:23Z
  generation: 0
  name: atcloud-wildcard
  namespace: ingress-nginx
  resourceVersion: "7396814"
  selfLink: /apis/certmanager.k8s.io/v1alpha1/namespaces/ingress-nginx/certificates/atcloud-wildcard
  uid: 91e878b8-6402-11e8-83f4-42010aa400df
spec:
  acme:
    config:
    - dns01:
        provider: google-dns
      domains:
      - '*.dev.k8.atcloud.io'
  commonName: ""
  dnsNames:
  - '*.dev.k8.atcloud.io'
  issuerRef:
    kind: ClusterIssuer
    name: letsencrypt-prod
  secretName: atcloud-wildcard-ingress-cert
status:
  acme:
    order:
      url: omitted
  conditions:
  - lastTransitionTime: 2018-06-03T15:54:32Z
    message: Certificate issued successfully
    reason: CertIssued
    status: "True"
    type: Ready
  - lastTransitionTime: 2018-06-03T15:54:30Z
    message: Order validated
    reason: OrderValidated
    status: "False"
    type: ValidateFailed
```

Why would istio-pilot be looking at/care about these?

**Comment by sakshigoel12** [comment]
@mandarjog @nmittler please could you take a look

**Comment by nmittler** [comment]
Tao, can you take a look? It appears to be related to cert rotation.

**Comment by wattli** [comment]
Seems pilot is trying to get *.dev.k8.atcloud.io periodically, and apparently that's not a valid service. @Stono , can you replace *.dev.k8.atcloud.io with some existing service? I am not a pilot expert, but I feel this may provide more debugging info.

And I remember @ldemailly / @costinm tried acme to issue k8s certs.

**Comment by wattli** [comment]
And this is more of a pilot issue than security (we only deal with istio cert, not k8s cert). Reassign to @costinm

**Comment by ldemailly** [comment]
costin is in vacation for 3 weeks so assigning to andra instead

**Comment by Stono** [comment]
> @Stono , can you replace *.dev.k8.atcloud.io with some existing service? I am not a pilot expert, but I feel this may provide more debugging info.

No, I cannot.  These resources are nothing to do with istio, and are used to facilitate SSL cert generation for our nginx ingress controller.  I can't change them because I need them as they are.  My confusion comes from what pilot is even looking at them, and seemingly considering them service accounts?

**Comment by nmittler** [comment]
FYI, while looking at another issue, I noticed the recent introduction of the call that may be resulting in this log: https://github.com/istio/istio/pull/5525/files#diff-f507fb55b7edf4a03d3c7e9aebf64a1eR94.

@rshriram @incfly I'm not sure why this would be showing up as a service, however .. any thoughts as to what might be going on?

**Comment by nmittler** [comment]
@incfly any thoughts?

**Comment by rshriram** [comment]
@nmittler is right. The error is coming from the code that validates service accounts. Somehow, the annotations are being converted into a service.

**Comment by incfly** [comment]
Sorry for the late response.

We previously make this call `GetIstioServiceAccount` in the authn plugin, and it [checks](https://github.com/istio/istio/pull/5525/files#diff-970c02a18802131086a418f813b9374eL342) whether it's a service outside the mesh before trying to set the validation context, I guess this explains why we have more invocations after the change?

**Comment by rshriram** [comment]
you mean the original dst cluster? Has the op added service entries for these services in the cluster?

**Comment by Stono** [comment]
@rshriram no I don't have service entries for those addresses

**Comment by emmayang** [comment]
I met same error, and the service cannot be found in my env is introduced into mesh via ServiceEntry.

**Comment by wansuiye** [comment]
I also met the same error, external service not be found via ServiceEntry, the service entry is:
```
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: redis
spec:
  hosts:
  - 10.20.30.40
  addresses:
  -10.20.30.40/32
  location: MESH_EXTERNAL
  ports:
  - number: 6379
    name: redis
    protocol: TCP
  resolution: NONE
---
```

**Comment by nmittler** [comment]
@incfly can you look at this? It's not clear that it's a user config error.

**Comment by ijsnellf** [comment]
I am pretty sure this is caused because of the aggregate service registry. By default, we always start up using an aggregate registry of a controller that monitors ServiceEntries and the regular K8S controller. When we make a call to GetServices(serviceX), the aggregate registry makes a call to the K8S registry and the ServiceEntry registry. See:

https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/aggregate/controller.go#L153

If serviceX is not in K8S, the K8S registry logs the warning:
https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/kube/controller.go#L248

**Comment by ijsnellf** [comment]
@nmittler @incfly ^^^

**Comment by Stono** [comment]
Hey, 
We're getting absolutely spammed with this log message for external services:

```
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: egress-internet-http
spec:
  hosts:
  - automotivemxin.com
  - www.google-analytics.com
  - google-analytics.com
  ports:
  - number: 80
    name: http
    protocol: HTTP
  - number: 443
    name: https
    protocol: HTTPS
```

Results in:
```
2018-07-23T12:01:35.008618Z     info    GetService(google-analytics.com) error: service does not exist
2018-07-23T12:01:35.008654Z     info    GetService(google-analytics.com) error: service does not exist
2018-07-23T12:01:35.011655Z     info    GetService(smtp.office365.com) error: service does not exist
2018-07-23T12:01:35.011699Z     info    GetService(smtp.office365.com) error: service does not exist
2018-07-23T12:01:35.011716Z     info    GetService(smtp.office365.com) error: service does not exist
2018-07-23T12:01:35.103425Z     info    GetService(*.airflow-worker.data-platform-airflow.svc.cluster.local) error: service does not exist
2018-07-23T12:01:35.203643Z     info    GetService(automotivemxin.com) error: service does not exist
2018-07-23T12:01:35.203728Z     info    GetService(automotivemxin.com) error: service does not exist
2018-07-23T12:01:35.504057Z     info    GetService(www.google-analytics.com) error: service does not exist
2018-07-23T12:01:35.504709Z     info    GetService(www.google-analytics.com) error: service does not exist
2018-07-23T12:01:35.606676Z     info    GetService(google-analytics.com) error: service does not exist
2018-07-23T12:01:35.606733Z     info    GetService(google-analytics.com) error: service does not exist
2018-07-23T12:01:35.609314Z     info    GetService(smtp.office365.com) error: service does not exist
2018-07-23T12:01:35.609443Z     info    GetService(smtp.office365.com) error: service does not exist
2018-07-23T12:01:35.609469Z     info    GetService(smtp.office365.com) error: service does not exist
2018-07-23T12:01:35.901031Z     info    GetService(*.airflow-worker.data-platform-airflow.svc.cluster.local) error: service does not exist
2018-07-23T12:01:35.906239Z     info    GetService(automotivemxin.com) error: service does not exist
2018-07-23T12:01:35.906501Z     info    GetService(automotivemxin.com) error: service does not exist
2018-07-23T12:01:36.001934Z     info    GetService(www.google-analytics.com) error: service does not exist
2018-07-23T12:01:36.003858Z     info    GetService(www.google-analytics.com) error: service does not exist
2018-07-23T12:01:36.007955Z     info    GetService(google-analytics.com) error: service does not exist
2018-07-23T12:01:36.008014Z     info    GetService(google-analytics.com) error: service does not exist
```

**Comment by smawson** [comment]
Assuming this was fixed in 1.0, please reopen if it is still a problem.

**Comment by Stono** [comment]
Yeah I'm not seeing it any more, thanks.
